server/tso/allocator_manager.go

```diff
@@ -327,11 +331,60 @@ func (am *AllocatorManager) updateAllocator(ag *allocatorGroup) {
 	}
 	if err := ag.allocator.UpdateTSO(); err != nil {
 		log.Warn("failed to update allocator's timestamp", zap.String("dc-location", ag.dcLocation), errs.ZapError(err))
-		ag.parentCancel()
+		am.resetAllocatorGroup(ag.dcLocation)
 		return
 	}
 }
 
+// Check if we have any new dc-location configured, if yes,
+// then set up the corresponding local allocator.
+func (am *AllocatorManager) allocatorPatroller(serverCtx context.Context) {
+	clusterDCLocations, err := am.GetClusterDCLocations()
+	if err != nil {
+		log.Error("check new allocators failed, can't get cluster dc-locations", errs.ZapError(err))
+	}
+	// Collect all dc-locations
+	dcLocations := make([]string, 0, len(clusterDCLocations))
+	for dcLocation := range clusterDCLocations {
+		dcLocations = append(dcLocations, dcLocation)
+	}
+	// Get all Local TSO Allocators
+	allocatorGroups := am.getAllocatorGroups(FilterDCLocation(config.GlobalDCLocation))
+	// Set up the new one
+	for _, dcLocation := range dcLocations {
+		if slice.NoneOf(allocatorGroups, func(i int) bool {
+			return allocatorGroups[i].dcLocation == dcLocation
+		}) {
+			if err := am.SetUpAllocator(serverCtx, dcLocation, election.NewLeadership(
```

Comment:
Can we gurantee that ts of the newly created allocator be greater than the last allocated global ts?

Reply:
Yes. When we synchronize the global ts and write it into each local allocator leader, we will also update the time window in etcd. So if the newly created allocator becomes a leader, it will be greater than the last allocated global ts.

Reply:
cool.